addon/-private/system/relationships/state/has-many.js

```diff
@@ -111,6 +111,26 @@ export default class ManyRelationship extends Relationship {
     super.removeCanonicalInternalModelFromOwn(internalModel, idx);
   }
 
+  removeDeletedInternalModelFromOwn(internalModel) {
```

Comment:
`removeInternalModelFromOwn` removes only current state while `removeDeletedInternalModelFromOwn` removes from current and canonical state.